### PR TITLE
Allow for CORS to be enabled on Reaper

### LINKED ISSUE #773: No way to enable CORS for Reaper
**What is missing?**

There is an environment variable that can be set on the Reaper container to enable CORS for the web server, there's no way to set that from a K8ssandraCluster manifest.

**Why do we need it?**

To allow for CORS requests to be completed successfully from a browser.

Related to https://github.com/thelastpickle/cassandra-reaper/pull/1244.

**Environment**

* K8ssandra Operator version:  1.4

**Anything else we need to know?**:  N/A

### PR DESCRIPTION
**Pull request by jdonenine**
**What this PR does**:
* Adds a flag `EnableCors` to allow for CORS to be enabled within Reaper.

**Which issue(s) this PR fixes**:
Fixes #773 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/reaper/deployment_test.go
Comment (burmanm) [inline pkg/reaper/deployment_test.go:288]: This is by default false (since it's not *bool), so testing the same thing as previous step. But it doesn't hurt.

**Comment by burmanm** [comment]
How is Karan blackmailing you to make PRs? ;)

The test failures don't seem to be related to this..

File: CHANGELOG/CHANGELOG-1.4.md
Comment (adejanovski) [inline CHANGELOG/CHANGELOG-1.4.md:18]: The changelog entry needs to go to `CHANGELOG-1.5.md` now.
We're only pushing bug fixes to the 1.4 branch ;)

**Review by adejanovski — CHANGES_REQUESTED** [review]
Aside from the changelog entry that needs to be moved to 1.5, LGTM 👍

File: CHANGELOG/CHANGELOG-1.4.md
Comment (adejanovski) [reply CHANGELOG/CHANGELOG-1.4.md:18]: @jdonenine, still waiting on this change until I can merge the PR 😉

Comment (jdonenine) [reply CHANGELOG/CHANGELOG-1.4.md:18]: My bad, I never saw this, i'll take care of it tonight.

**Comment by jdonenine** [comment]
OK @adejanovski when I said I'd do it the other night I really meant a week later 😬 but it's rebased and updated the newer changelog.

**Comment by jdonenine** [comment]
😔 I need to look at that test failure

**Comment by adejanovski** [comment]
Closing as stale.

**Pull request closed without merging**
